nucypher/blockchain/eth/agents.py

```diff
@@ -55,13 +56,20 @@ class ContractAgency:
     @classmethod
     def get_agent(cls,
                   agent_class,
-                  registry: BaseContractRegistry,
+                  registry: BaseContractRegistry = None,
                   provider_uri: str = None,
                   ) -> 'EthereumContractAgent':
 
         if not issubclass(agent_class, EthereumContractAgent):
             raise TypeError(f"Only agent subclasses can be used from the agency.")
-        registry_id = registry.id
+
+        if not registry:
+            if len(cls.__agents) == 1:
+                registry = list(cls.__agents.keys()).pop()
```

Comment:
Woah - what is going on here?

Reply:
This is just to make the registry parameter optional, only when there's already an existing registry being tracked by the agency. I'm taking the same approach than `BlockchainInterfaceFactory.get_interface()`, which uses a similar trick to make the `provider_uri` optional.